steel/src/lib.rs

```diff
@@ -179,3 +128,72 @@ pub trait EvmBlockHeader: Sealable {
     /// Fills the EVM block environment with the header's data.
     fn fill_block_env(&self, blk_env: &mut BlockEnv);
 }
+
+// Keep everything in the Steel library private except the commitment.
+mod private {
+    alloy_sol_types::sol! {
+        #![sol(all_derives)]
+        /// A commitment to a specific block in the blockchain.
+        struct Commitment {
+            /// Encodes both the block identifier (block number or timestamp) and the version.
+            uint256 blockID;
+            /// The block hash or beacon block root, used for validation.
+            bytes32 blockDigest;
+        }
+    }
+}
+
+/// Solidity struct representing the committed block used for validation.
+pub use private::Commitment as SolCommitment;
```

Comment:
Since there is not another type named `Commitment`, I would lean towards making this publicly available with that name rather than `SolCommitment`. Based on this name, I figured this was the "Solidity version" of some other Rust type and was surprised not to find that other type anywhere.

Reply:
Yeah, I see what you mean. I thought it might make sense to use this Sol* name to indicate that the struct has been generated using the `sol!` macro, as for those the type and especially the naming could be strange. It is a shame that the macro does not convert snake_case to camelCase automatically.

Reply:
However, I don't have a strong opinion on this at all. I am fine with removing this renaming if we think it makes the naming cleaner. I am also not a fan of the `mod private`, as it is not really necessary.

Reply:
I went ahead and renamed `SolCommitment` to `Commitment`. I did not remove the `private` mod as I just wasn't sure if there are perhaps more symbols in that module that would get exported.